src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -185,11 +185,20 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
 
         protected override async Task<bool> HandleUnauthorizedAsync(ChallengeContext context)
         {
-            var authResult = await HandleAuthenticateOnceAsync();
+            Exception failure = null;
```

Comment:
There's once thing I'd to bring into this discussion.

`HandleAuthenticateAsync` should be treated same as `HandleAuthenticateOnceAsync`, because later is just former with a restriction of not to be repeatedly invoked. 

So if we set up the rule that `HandleAuthenticateOnceAsync` won't throw exception the rule should apply to `HandleAuthenticateAsync`, too. This is not the case in `JwtBearerHandler` today ([see here](https://github.com/aspnet/Security/blob/137e18f39d6f9e0932916fa9e3a8c296e8dd8978/src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs#L182)).

It is actually unavoidable because `HandleAutneticateAsync` is virtual.

Should we consider a template pattern to enforce this rule? (change will be out of the scope of this issue, but worth considering).